File: docs/qa/build-reteste-nomade.mjs
import fs from 'node:fs/promises';
import { Workbook, SpreadsheetFile } from '@oai/artifact-tool';

const outputDir = String.raw`C:\Users\Teste\Documentos 2.0\Verificação 2.0\_DESENVOLVIMENTOS\Plataforma allka 2026\allka-plataforma\allka-2026\entregas\qa-usuarios\atualizacao-nomade`;
await fs.mkdir(outputDir, { recursive: true });

const tests = [
  ['NO-R01','Login e sessão','Acesse com a conta Nômade de QA.','Portal Nômade abre e a sessão permanece ativa.','Baixo'],
  ['NO-R02','Container padrão','Confira cabeçalho, menu e conteúdo central.','Nenhuma área cortada ou sobreposta.','Baixo'],
  ['NO-R03','Tutorial no primeiro acesso','Caso a oferta apareça, clique em Começar.','Primeiro passo do tutorial inicia corretamente.','Baixo'],
  ['NO-R04','Não quero ver novamente','Escolha Não quero ver e recarregue a página.','Oferta não reaparece sozinha; tour continua na Ajuda.','Baixo'],
  ['NO-R05','Central de Ajuda','Abra Ajuda e localize o tour do Nômade.','Ícone é diferente de Sugestões; iniciar, continuar e refazer funcionam.','Baixo'],
  ['NO-R06','Retomada do tutorial','Avance dois passos, feche e recarregue.','Tour continua no mesmo ponto.','Baixo'],
  ['NO-R07','Allkademy','Abra Allkademy pelo menu.','Layout segue o padrão com menu, cabeçalho e container central.','Baixo'],
  ['NO-R08','Habilitações','Abra Habilitações e confira as áreas.','Tela abre sem erro e explica as habilitações.','Baixo'],
  ['NO-R09','Minhas Tarefas em lista','Abra Minhas Tarefas e selecione Lista.','Cards têm prazo, etapa e estado legíveis.','Baixo'],
  ['NO-R10','Minhas Tarefas em kanban','Alterne para Kanban e volte para Lista.','As duas visualizações funcionam.','Baixo'],
  ['NO-R11','Detalhe da tarefa','Clique em tarefa ou etapa atribuída.','Abre no container central, com fechar e fixar.','Baixo'],
  ['NO-R12','Etapa concluída','Em registro [TESTE USUÁRIO], conclua a etapa liberada e abra Concluídas.','Etapa concluída continua visível em Concluídas.','Médio'],
  ['NO-R13','Sequência automática','Em tarefa [TESTE USUÁRIO] com duas etapas, conclua a primeira.','Próxima etapa é liberada; antes disso, explica o bloqueio.','Médio'],
  ['NO-R14','Histórico','Abra Histórico após uma ação permitida.','Ação aparece com data e descrição coerentes.','Baixo'],
  ['NO-R15','Tarefas Disponíveis','Abra Tarefas Disponíveis e assuma apenas uma etapa [TESTE USUÁRIO] compatível.','Busca funciona e etapa assumida aparece em Minhas Tarefas.','Médio'],
  ['NO-R16','Permissões do Nômade','Procure Catálogo ou Contratar no menu.','Não há catálogo nem ações de compra.','Baixo'],
];

const wb = Workbook.create();
const summary = wb.worksheets.add('Resumo');
const sheet = wb.worksheets.add('Reteste Nômade');
for (const ws of [summary, sheet]) ws.showGridLines = false;

summary.getRange('A2').values = [['Reteste Nômade']];
summary.getRange('A3').values = [['Conta: nomad@allka.com.vc']];
summary.getRange('A5:B9').values = [
  ['Status','Quantidade'],
  ['Funcionou',0],
  ['Não funcionou',0],
  ['Não se aplica',0],
  ['Não consegui testar',0],
];
summary.getRange('B6').formulas = [[`=COUNTIF('Reteste Nômade'!$G$2:$G$17,A6)`]];
summary.getRange('B6:B9').fillDown();
summary.getRange('A11').values = [['Observação: NO-R12 a NO-R15 exigem registros [TESTE USUÁRIO] preparados pela coordenação.']];

const headers = [['Código','Área','Como testar','Resultado esperado','Cuidado','Evidência ou URL','Status','Observações']];
sheet.getRange('A1:H1').values = headers;
sheet.getRange(`A2:H${tests.length + 1}`).values = tests.map(([code, area, how, expected, care]) => [code, area, how, expected, care, '', '', '']);

const titleFormat = { font: { name: 'Arial', size: 16, bold: true, color: '#111827' } };
summary.getRange('A2').format = titleFormat;
summary.getRange('A3').format = { font: { name: 'Arial', size: 10, italic: true, color: '#475569' } };
summary.getRange('A5:B5').format = { fill: '#14213D', font: { name: 'Arial', size: 10, bold: true, color: '#FFFFFF' } };
summary.getRange('A6:B9').format = { font: { name: 'Arial', size: 10 }, borders: { items: [{ color: '#D9D9D9', style: 'continuous', side: 'EdgeBottom' }] } };
summary.getRange('A11').format = { font: { name: 'Arial', size: 10, italic: true, color: '#7C2D12' }, fill: '#FFF7ED', wrapText: true };
summary.getRange('A1:B12').format.horizontalAlignment = 'left';
summary.getRange('A1:B12').format.verticalAlignment = 'center';
summary.getRange('A:A').format.columnWidth = 28;
summary.getRange('B:B').format.columnWidth = 16;
summary.getRange('A11:B11').merge();

sheet.getRange('A1:H1').format = { fill: '#14213D', font: { name: 'Arial', size: 10, bold: true, color: '#FFFFFF' }, horizontalAlignment: 'center', verticalAlignment: 'center', wrapText: true };
sheet.getRange(`A2:H${tests.length + 1}`).format = { font: { name: 'Arial', size: 10, color: '#172033' }, verticalAlignment: 'top', wrapText: true };
sheet.getRange(`A2:H${tests.length + 1}`).format.rowHeight = 52;
sheet.getRange('A:A').format.columnWidth = 12;
sheet.getRange('B:B').format.columnWidth = 25;
sheet.getRange('C:C').format.columnWidth = 44;
sheet.getRange('D:D').format.columnWidth = 44;
sheet.getRange('E:E').format.columnWidth = 12;
sheet.getRange('F:F').format.columnWidth = 25;
sheet.getRange('G:G').format.columnWidth = 22;
sheet.getRange('H:H').format.columnWidth = 35;
sheet.getRange(`G2:G${tests.length + 1}`).dataValidation = { rule: { type: 'list', formula1: '"Funcionou,Não funcionou,Não se aplica,Não consegui testar"' } };
sheet.getRange(`G2:G${tests.length + 1}`).conditionalFormats.add('containsText', { text: 'Funcionou', format: { fill: '#DCFCE7', font: { color: '#166534', bold: true } } });
sheet.getRange(`G2:G${tests.length + 1}`).conditionalFormats.add('containsText', { text: 'Não funcionou', format: { fill: '#FEE2E2', font: { color: '#991B1B', bold: true } } });
sheet.getRange(`G2:G${tests.length + 1}`).conditionalFormats.add('containsText', { text: 'Não se aplica', format: { fill: '#F1F5F9', font: { color: '#475569' } } });
sheet.getRange(`G2:G${tests.length + 1}`).conditionalFormats.add('containsText', { text: 'Não consegui testar', format: { fill: '#FEF3C7', font: { color: '#92400E', bold: true } } });
sheet.freezePanes.freezeRows(1);
sheet.tables.add(`A1:H${tests.length + 1}`, true, 'RetesteNomadeTable').style = 'TableStyleMedium2';

wb.recalculate();
const preview = await wb.render({ sheetName: 'Reteste Nômade', range: 'A1:H17', autoCrop: 'all', scale: 1, format: 'png' });
await fs.writeFile(`${outputDir}\\Registro_de_Resultados_Reteste_Nomade_preview.png`, new Uint8Array(await preview.arrayBuffer()));
const xlsx = await SpreadsheetFile.exportXlsx(wb);
await xlsx.save(`${outputDir}\\Registro_de_Resultados_Reteste_Nomade.xlsx`);
console.log('done');

File: docs/qa/build-reteste-nomade.py
from pathlib import Path
from docx import Document
from docx.shared import Inches, Pt, RGBColor
from docx.enum.text import WD_ALIGN_PARAGRAPH
from docx.enum.table import WD_TABLE_ALIGNMENT, WD_CELL_VERTICAL_ALIGNMENT
from docx.oxml import OxmlElement
from docx.oxml.ns import qn

ROOT = Path(r"C:\Users\Teste\Documentos 2.0\Verificação 2.0\_DESENVOLVIMENTOS\Plataforma allka 2026\allka-plataforma\allka-2026")
OUT = ROOT / "entregas" / "qa-usuarios" / "atualizacao-nomade"
OUT.mkdir(parents=True, exist_ok=True)

tests = [
    ("NO-R01", "Login e sessão", "Acesse https://allka.store com a conta Nômade de QA.", "O login abre o portal do Nômade e a sessão permanece ativa ao navegar."),
    ("NO-R02", "Container padrão", "Na tela inicial, confira cabeçalho, menu lateral e conteúdo central.", "Nenhuma área fica cortada ou sobreposta. O conteúdo usa o mesmo container central da plataforma."),
    ("NO-R03", "Tutorial no primeiro acesso", "Caso a oferta apareça, clique em Começar.", "O primeiro passo do tutorial abre e o balão destaca o elemento correto."),
    ("NO-R04", "Não quero ver novamente", "Na oferta do tutorial, escolha Não quero ver mais este tutorial e recarregue a página.", "A oferta não reaparece sozinha após F5. O tutorial continua disponível manualmente pela Ajuda."),
    ("NO-R05", "Central de Ajuda", "Abra o ícone de Ajuda e localize o tutorial do portal Nômade.", "O ícone de Ajuda é diferente do ícone de Sugestões. É possível iniciar, continuar ou refazer o tour."),
    ("NO-R06", "Retomada do tutorial", "Inicie um tour, avance dois passos, feche e recarregue a página.", "O tour oferece continuidade no mesmo ponto; não volta silenciosamente ao início."),
    ("NO-R07", "Allkademy", "Abra Allkademy pelo menu do Nômade.", "A tela abre com cabeçalho, menu lateral e container central padronizados."),
    ("NO-R08", "Habilitações", "Abra Habilitações e confirme as áreas disponíveis para o perfil de QA.", "A tela abre sem erro e explica claramente as habilitações. Registre se as áreas necessárias para os testes foram configuradas."),
    ("NO-R09", "Minhas Tarefas em lista", "Abra Minhas Tarefas e selecione Lista.", "As tarefas atribuídas aparecem como cards legíveis, com prazo, etapa e estado."),
    ("NO-R10", "Minhas Tarefas em kanban", "Na mesma tela, selecione Kanban e volte para Lista.", "As duas visualizações funcionam e mostram a mesma situação das tarefas."),
    ("NO-R11", "Detalhe da tarefa", "Clique em uma tarefa ou etapa atribuída.", "O detalhe abre dentro do container central, no tamanho da área principal, com fechar e fixar na bandeja de telas."),
    ("NO-R12", "Etapa concluída", "Em uma tarefa marcada [TESTE USUÁRIO], conclua apenas a etapa liberada e abra a aba Concluídas.", "A tarefa/etapa concluída permanece visível na aba Concluídas, sem desaparecer."),
    ("NO-R13", "Sequência automática", "Em uma tarefa [TESTE USUÁRIO] com duas etapas sequenciais, conclua a primeira.", "A próxima etapa é liberada automaticamente. Antes disso, ela aparece bloqueada e explica o motivo."),
    ("NO-R14", "Histórico", "Abra Histórico após uma ação de teste permitida.", "A ação aparece com data e descrição coerentes."),
    ("NO-R15", "Tarefas Disponíveis", "Abra Tarefas Disponíveis. Se existir uma etapa [TESTE USUÁRIO] compatível com uma habilitação, assuma-a.", "A busca filtra normalmente. Após assumir uma etapa de teste, ela passa a aparecer em Minhas Tarefas."),
    ("NO-R16", "Permissões do Nômade", "Percorra o menu e tente localizar catálogo de contratação.", "O Nômade não vê catálogo nem ações de compra. Só aparecem recursos próprios do perfil."),
]

def set_cell_shading(cell, fill):
    tcPr = cell._tc.get_or_add_tcPr()
    shd = OxmlElement('w:shd')
    shd.set(qn('w:fill'), fill)
    tcPr.append(shd)

def set_cell_margins(cell, top=80, start=100, bottom=80, end=100):
    tc = cell._tc
    tcPr = tc.get_or_add_tcPr()
    tcMar = tcPr.first_child_found_in('w:tcMar')
    if tcMar is None:
        tcMar = OxmlElement('w:tcMar')
        tcPr.append(tcMar)
    for m, val in [('top', top), ('start', start), ('bottom', bottom), ('end', end)]:
        node = tcMar.find(qn(f'w:{m}'))
        if node is None:
            node = OxmlElement(f'w:{m}')
            tcMar.append(node)
        node.set(qn('w:w'), str(val))
        node.set(qn('w:type'), 'dxa')

def style_table(table):
    table.alignment = WD_TABLE_ALIGNMENT.CENTER
    table.style = 'Table Grid'
    for idx, cell in enumerate(table.rows[0].cells):
        set_cell_shading(cell, '14213D')
        cell.vertical_alignment = WD_CELL_VERTICAL_ALIGNMENT.CENTER
        for run in cell.paragraphs[0].runs:
            run.font.bold = True
            run.font.color.rgb = RGBColor(255, 255, 255)
            run.font.size = Pt(9)
    for row_i, row in enumerate(table.rows[1:], start=1):
        for cell in row.cells:
            cell.vertical_alignment = WD_CELL_VERTICAL_ALIGNMENT.CENTER
            set_cell_margins(cell)
            for run in cell.paragraphs[0].runs:
                run.font.size = Pt(9)
        if row_i % 2 == 0:
            for cell in row.cells:
                set_cell_shading(cell, 'F4F7FB')

doc = Document()
section = doc.sections[0]
section.top_margin = Inches(0.65)
section.bottom_margin = Inches(0.6)
section.left_margin = Inches(0.7)
section.right_margin = Inches(0.7)
styles = doc.styles
styles['Normal'].font.name = 'Arial'
styles['Normal']._element.rPr.rFonts.set(qn('w:ascii'), 'Arial')
styles['Normal']._element.rPr.rFonts.set(qn('w:hAnsi'), 'Arial')
styles['Normal'].font.size = Pt(10)
for name, size in [('Title', 22), ('Heading 1', 15), ('Heading 2', 12)]:
    styles[name].font.name = 'Arial'
    styles[name]._element.rPr.rFonts.set(qn('w:ascii'), 'Arial')
    styles[name]._element.rPr.rFonts.set(qn('w:hAnsi'), 'Arial')
    styles[name].font.size = Pt(size)
    styles[name].font.color.rgb = RGBColor(0, 0, 0)

title = doc.add_paragraph(style='Title')
title.alignment = WD_ALIGN_PARAGRAPH.LEFT
title.add_run('Roteiro de Reteste Nômade')
subtitle = doc.add_paragraph()
subtitle.add_run('Atualização de QA - setembro de 2026').italic = True
intro = doc.add_paragraph()
intro.add_run('Objetivo. ').bold = True
intro.add_run('Este roteiro verifica as correções recentes do portal Nômade antes de liberar o fluxo para o uso normal. Execute apenas com a conta de QA indicada e registre cada resultado na planilha correspondente.')

doc.add_heading('Acesso de teste', level=1)
t = doc.add_table(rows=1, cols=2)
t.rows[0].cells[0].text = 'Campo'
t.rows[0].cells[1].text = 'Informação'
for a, b in [
    ('Portal', 'https://allka.store'),
    ('Conta', 'nomad@allka.com.vc'),
    ('Perfil', 'Nômade de QA, ativo, nível Bronze'),
    ('Senha', 'Enviar separadamente por canal interno seguro'),
]:
    row = t.add_row().cells
    row[0].text = a
    row[1].text = b
style_table(t)

doc.add_heading('Antes de começar', level=1)
for text in [
    'Use apenas registros marcados [TESTE USUÁRIO] para concluir etapas ou assumir tarefas.',
    'Os testes NO-R12 a NO-R15 precisam de uma tarefa de QA atribuída e de uma tarefa disponível compatível. Se elas não existirem, marque Não consegui testar e informe a coordenação.',
    'Não exclua, cancele ou altere dados reais. Não publique produtos, não faça pagamentos e não altere contas.',
    'Ao encontrar algo errado, tire uma captura de tela, informe o código do teste e descreva o que esperava acontecer.',
]:
    p = doc.add_paragraph(style='List Bullet')
    p.add_run(text)

doc.add_heading('Roteiro de testes', level=1)
for code, name, steps, expected in tests:
    doc.add_heading(f'{code} {name}', level=2)
    p = doc.add_paragraph()
    p.add_run('Como testar: ').bold = True
    p.add_run(steps)
    p = doc.add_paragraph()
    p.add_run('Deve acontecer: ').bold = True
    p.add_run(expected)

doc.add_heading('Como registrar', level=1)
doc.add_paragraph('Na planilha, selecione Funcionou, Não funcionou, Não se aplica ou Não consegui testar. Em caso de problema, descreva o que ocorreu, inclua a URL da tela e anexe a captura quando possível.')
doc.add_heading('Pendências conhecidas fora deste reteste', level=1)
doc.add_paragraph('Em telas estreitas, os ícones flutuantes da lateral direita podem ficar parcialmente atrás do container principal. Esse acabamento já foi identificado e não faz parte deste reteste; registre apenas se o comportamento piorar ou impedir uma ação principal.')

footer = section.footer.paragraphs[0]
footer.alignment = WD_ALIGN_PARAGRAPH.CENTER
footer.add_run('Allka - roteiro interno de QA')

out_file = OUT / 'Roteiro_de_Reteste_Nomade.docx'
doc.save(out_file)
print(out_file)
